jvm-diagnostics/src/main/java/com/palantir/jvm/diagnostics/virtualthreads/VirtualThreadsDiagnostics.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.jvm.diagnostics.virtualthreads;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.lang.management.ManagementFactory;
+import java.lang.management.PlatformManagedObject;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Provides a runtime delegate wrapper around various JMX-based diagnostics for virtual threads.
+ * <p>
+ * Currently, this implementation includes wrappers for the following:
+ * <ul>
+ *     <li><a href="https://docs.oracle.com/en/java/javase/24/docs/api/jdk.management/jdk/management/VirtualThreadSchedulerMXBean.html">jdk.management.VirtualThreadSchedulerMXBean</a> (available on JDK24+)</li>
+ * </ul>
+ */
+@SuppressWarnings("AbbreviationAsWordInName")
+public class VirtualThreadsDiagnostics {
```

Comment:
I notice that we expose other JVM diagnostics by creating an `Accessor` interface (similar to `VirtualThreadSchedulerMXBeanSupport` defined locally here), and implementing it in the JvmDiagnostics.java file. We could use it in tritium like we have for [DNS cache metrics.](https://github.com/palantir/tritium/blob/00541bc3e4ebc15518d6ecbee9ecb274f931ffb7/tritium-metrics-jvm/src/main/java/com/palantir/tritium/metrics/jvm/JvmMetrics.java#L182-L194)

What are your thoughts on following that pattern? [I sketched my idea here.](https://github.com/palantir/jvm-diagnostics/compare/blaub/jmx-virtual-threads...mpritham/virt-thread-proposal?expand=1) That branch is missing tests, and there are a few log statements that I haven't included: just using it to show what I mean structurally.

Reply:
yeah that feels directionally right i suppose. let me try to adjust my change and see how it looks.

Reply:
alright, i've refactored this in dd49ed1d9133f4ba1dce36b1faee0ab3d979813b